cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-health-check.adoc

```diff
@@ -0,0 +1,68 @@
+:context: cloud-experts-deploying-application-health-check
+
+toc::[]
+
+//rosaworkshop.io content metadata
+//Brought into ROSA product docs 2024-01-26
+
+== Health checks
+You can intentionally crash your pod making it unresponsive to the Kubernetes liveness probes to see how Kubernetes responds to the event. We will first intentionally crash our pod and see that Kubernetes will self-heal by immediately spinning it back up. Then we will trigger the health check by stopping the response on the `/health` endpoint in our application. After three consecutive failures Kubernetes should stop the pod and then re-create it.
+
+=== Preparing the desktop
+
+. Split your desktop screen between the OpenShift web console and the OSToy application so that you can see the results of your actions immediately.
++
+image::5-ostoy-splitscreen.png[Splitscreen desktop with the OSToy application and the web console]
++
+If your screen is too small or you cannot split your screen for another reason, open the OSToy application in another tab so you can quickly switch to the OpenShift web console after activating the features in the application. 
+
+. From the web console, select *Workloads > Deployments > ostoy-frontend* to view the OSToy deployment. 
++
+image::5-ostoy-deployview.png[The web console deployments page]
+
+==== Crashing the pod 
+
+. From the OSToy application, click *Home* in the left menu, and enter a message in the *Crash Pod* tile (for example, "_This is goodbye!_"). 
+
+. Click *Crash Pod*.
++
+image::5-ostoy-crashpod.png[OSToy crash pod selection]
++
+The pod crashes and Kubernetes should restart the pod:
++
+image::5-ostoy-crashmsg.png[OSToy pod crash message]
+
+==== Viewing the revived pod
+
+. From the web console, quickly switch to the *Deployments* screen. You will see that the pod turns yellow, meaning it is down but should quickly revive and turn blue. The revival process happens quickly so you might miss it.
++
+image::5-ostoy-podcrash.gif[Deployment details page]
+
+==== 4. Confirmation
+You can check the pod events. 
+
+. From the web console, click  *Pods > ostoy-frontend-xxxxxxx-xxxx* to change to the pods screen.
++
+image::5-ostoy-events.png[Pod overview page]
+
+. Click *Events* sub tab and verify that the container has crashed and restarted.
++
+image::5-ostoy-podevents.png[Pod events list]
+
+==== Making the application malfunction
+
+Keep the pod events page open from the previous procedure. 
+
+* From the OSToy application, click *Toggle Health* in the *Toggle Health Status* tile. You will see *Current Health* switch to *I'm not feeling all that well*.
++
+image::5-ostoy-togglehealth.png[OSToy toggle health tile]
+
+.Verification
+
+After the previous step, the application stops responding with a `200 HTTP code`. After 3 consecutive failures *A*, Kubernetes will stop the pod *B* and restart it *C*. From the web console, switch back to the pod events page and you will see that the liveness probe failed and the pod restarted.
```

Comment:
Can we change the bolded letters to callouts so they don't get lost in the bulk of the text?

Reply:
got it.